cylc/flow/commands.py

```diff
@@ -144,45 +141,61 @@ def _command(fcn: '_TCommand') -> '_TCommand':
     return _command
 
 
+def _report_unmatched(unmatched: Set[TaskTokens]):
```

Comment:
Why not build this in to `id_match`?

Reply:
We don't always want to report unmatched tasks (e.g, we have some internal uses for this interface). And we might want to change the message / reporting on a command by command basis.

I've centralised the part of the code which handles user input  in one place, and not leaked its logic elsewhere.